source/app/models/url.rb

```diff
@@ -0,0 +1,18 @@
+class Url < ActiveRecord::Base
+  validates :original, presence: true
+  validates :original, length: { minimum: 1 }
+  validates :original, format: {with: /\A((http|https):\/\/).*\Z/}
+
+
+  before_save :shortify
+
+  private
+  def shortify
+    if self.shortened == nil
+      self.click_count = 0
+      begin
+        self.shortened = "https://bit.ly/" + SecureRandom.hex(3)
```

Comment:
Extremely minor, nitpicky detail but 3 hex digits would only be 4096 possibly urls. Surely there are more than that. From my research, Base62 is commonly used which is `a-z`, `A-Z` and `0-9`. Three base62 digits would give 238,328 urls by comparison.